index.bs

```diff
@@ -1349,15 +1350,31 @@ Canvas Rendering Context {#canvas-rendering-context}
 XRPresentationContext {#xrpresentationcontext-interface}
 ---------------------
 
+When the {{HTMLCanvasElement/getContext()}} method of an {{HTMLCanvasElement}} |canvas| is to return a new object for the <var ignore>contextId</var> <code>present-xr</code>, the user agent must return an {{XRPresentationContext}} with {{XRPresentationContext/canvas}} set to |canvas|.
+
 <pre class="idl">
 [SecureContext, Exposed=Window] interface XRPresentationContext {
   readonly attribute HTMLCanvasElement canvas;
 };
 </pre>
 
-Each {{XRPresentationContext}} has an associated <dfn attribute for="XRPresentationContext">canvas</dfn> which is an {{HTMLCanvasElement}} set during creation.
+The <dfn attribute for="XRPresentationContext">canvas</dfn> attribute indicates the {{HTMLCanvasElement}} that created this context.
+
+Each {{XRPresentationContext}} has an <dfn for="XRPresentationContext">session</dfn> which is an {{XRSession}} initially set to <code>null</code>.
+
+<div class="algorithm" data-algorithm="presentation-context-session-update">
+
+When another algorithm indicates it will <dfn>update the XRPresentationContext session</dfn> to {{XRSession}} |session|, run the following steps:
+
+  1. Let |context| be the target {{XRPresentationContext}}.
+  1. Let |prevSession| be |context|'s [=XRPresentationContext/session=].
+  1. If |prevSession| is equal to |session| abort these steps.
+  1. If |prevSession| is not <code>null</code>:
+    1. If |prevSession|'s {{XRSession/renderState}}'s {{XRRenderState/outputContext}} is equal to |context| set |prevSession|'s {{XRSession/renderState}}'s {{XRRenderState/outputContext}} to <code>null</code>.
```

Comment:
Does this need to be queued up for processing by |prevSession|?

Reply:
It should take effect immediately, IMO. Otherwise you could feasibly go for a full frame with two different sessions both reporting that they are using the same context.

Reply:
I agree that a frame in which two sessions both believe they're using the same context would be bad.... my bigger concern is that two different sessions might not be on the same thread if OffscreenCanvas is used and that as a result this change could cause a problem for `|prevSession|`

Reply:
Ha! Okay! I knew if I procrastinated on this long enough the answer would come to me... 😅

You can't use an object in more than one worker at a time. Javascript only allows for them to be transferred between workers. (Exception is shared arrays, which only contain binary data, not objects.) AND FURTHERMORE you have to explicitly mark an object as transferable in order to move it between workers and we have not done so with this context type, SO......

It's a non-issue! YAY! 🎉

Reply:
Hah!  Ok, that's fair.